CRM/Financial/BAO/FinancialItem.php

```diff
@@ -286,25 +286,25 @@ public static function checkContactPresent($contactIds, &$error) {
   }
 
   /**
-   * Get last financial item data.
+   * Get most relevant previous financial item relating to the line item.
    *
-   * @param int $entityId
+   * This function specifically excludes sales tax.
    *
-   * @param string $entityTable
+   * @param int $entityId
    *
    * @return object CRM_Core_DAO
    */
-  public static function getPreviousFinancialItem($entityId, $entityTable = 'civicrm_line_item') {
-    $queryParams = array(
-      1 => array($entityId, 'Integer'),
-      2 => array($entityTable, 'String'),
+  public static function getPreviousFinancialItem($entityId) {
+    $params = array(
+      'entity_id' => $entityId,
+      'entity_table' => 'civicrm_line_item',
```

Comment:
This issue is this table should only be called from core for a specific purpose - if we are expanding that purpose we should do it with the addition of unit tests and ensure that the name & docblocks are fair. You said it might be used from an extension and I see that as a reason NOT to support anything more than a specific entityTable - we don't want to support non-core usage of a function like this